openpiv/PIV_3D_plotting.py

```diff
@@ -119,32 +123,80 @@ def plot_3D_alpha(data):
     plt.show()
 
 
-def quiver_3D(u, v, w, x=None, y=None, z=None, image_dim=None, mask_filtered=None, filter_def=0, filter_reg=(1, 1, 1),
-              cmap="jet", quiv_args={}, cbound=None):
-    # filter_def filters values with smaler absolute deformation
-    # nans are also removed
-    # setting the filter to <0 will probably mess up the arrow colors
-    # filter_reg filters every n-th value, separate for x, y, z axis
-    # you can also provide your own mask with mask_filtered !!! make sure to filter out arrows with zero total deformation!!!!
-    # other wise the arrows are not colored correctly
-    # use indices for x,y,z axis as default - can be specified by x,y,z
+def quiver_3D(u, v, w, x=None, y=None, z=None, mask_filtered=None, filter_def=0, filter_reg=(1, 1, 1),
+              cmap="jet", quiv_args=None, vmin=None, vmax=None):
+    """ Displaying 3D deformation fields vector arrows
+
+       Parameters
+       ----------
+        u,v,w: 3d ndarray or lists
+            arrays or list with deformation in x,y and z direction
+
+        x,y,z: 3d ndarray or lists
+             Arrays or list with deformation the coordinates of the deformations.
+             Must match the dimensions of the u,v qnd w. If not provided x,y and z are created
+             with np.indices(u.shape)
+
+        mask_filtered, boolean 3d ndarray or 1d ndarray
+             Array, or list with same dimensions as the deformations. Defines the area where deformations are drawn
+        filter_def: float
+             Filter that prevents the display of deformations arrows with length < filter_def
+        filter_reg: tuple
+             Filter that prevents the display of every i-th deformations arrows separatly alon each axis.
+             filter_reg=(2,2,2) means that only every second arrow along x,y z axis is displayed leading to
+             a total reduction of displayed arrows by a factor of 8.
+        cmap: string
+             matplotlib colorbar that defines the coloring of the arrow
+        quiv_args: dict
+            Dictionary with kwargs passed on to the matplotlib quiver function.
+
+        vmin,vmax: float
+            Upper and lower bounds for the colormap. Works like vmin and vmax in plt.imshow().
+
+       Returns
+       -------
+        fig: matploltib figure object
+
+        ax: mattplotlib axes object
+            the holding the main 3D quiver plot
+
+       """
 
     # default arguments for the quiver plot. can be overwritten by quiv_args
     quiver_args = {"normalize": False, "alpha": 0.8, "pivot": 'tail', "linewidth": 1, "length": 20}
-    quiver_args.update(quiv_args)
-    if not isinstance(image_dim, (list, tuple, np.ndarray)):
-        image_dim = np.array(u.shape)
+    if isinstance(quiv_args, dict):
+        quiver_args.update(quiv_args)
 
+    # generating coordinates if not provided
```

Comment:
this probably should be a general function somewhere as these lines are repeated several times, right?